build_runner/lib/src/asset_graph/node.dart

```diff
@@ -169,6 +166,41 @@ class GeneratedAssetNode extends AssetNode {
       'GeneratedAssetNode: $id generated from input $primaryInput.';
 }
 
+/// A typical generated node which was generated in a specific phase, and used a
+/// set of [globs] during its generation.
+class GeneratedForPhaseAssetNode extends GeneratedAssetNode {
```

Comment:
My Java days have trained me to be scared of extending a non-abstract class. Should we consider a separate sub-type for post-build outputs?

Reply:
So actually I may be able to revert this part - I am testing out just making `BuildAction` not have a `builder` field, and instead there would be a `BuilderBuildAction` and `PostBuildAction`.

That would allow me to actually give a `phaseNumber` to the post build actions and get some other stuff for free as well.

It turns out we don't reference the `builder` field from `BuildAction` all that much so adding the type checks where necessary shouldn't be difficult, and it would allow me to provide an `Iterable<PostProcessBuilder> builders` field for the `PostBuildAction`.